src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,179 +1,280 @@
+    private var inInventory = false
 
     @SubscribeEvent
-    fun onRenderItemTip(event: RenderItemTipEvent) {
-        if (!LorenzUtils.inSkyBlock || LorenzUtils.skyBlockArea != "Dungeon Hub") return
-        if (!config.floorAsStackSize) return
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        val inventoryName = event.inventoryName
 
-        val itemName = event.stack.name.removeColor()
-        val invName = InventoryUtils.openInventoryName()
+        DelayedRun.runDelayed(10.0.milliseconds) {
+            floorStackSize.clear()
+            highlightParty.clear()
+//
+//        Code for Stack Tip
+//
+            if (catacombsGatePattern.matches(inventoryName)) {
+                val lore = event.inventoryItems[45]?.getLore()
+                inInventory = true
+                if (lore != null) {
+                    if (lore.size > 3 && lore[0] == "§7View and select a dungeon class.") {
+                        selectedClass = lore[2].split(" ").last().removeColor()
+                    }
+                }
 
-        if (invName == "Select Floor") {
-            if (itemName == "Any") {
-                event.stackTip = "A"
-            } else if (itemName == "Entrance") {
-                event.stackTip = "E"
-            } else if (itemName.startsWith("Floor ")) {
-                event.stackTip = itemName.split(' ').last().romanToDecimalIfNecessary().toString()
-            }
-        } else if (itemName.startsWith("The Catacombs - ") || itemName.startsWith("MM Catacombs -")) {
-            val floor = itemName.split(" - ").last().removeColor()
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = itemName.contains("MM ")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
-            }
-        } else if (itemName.endsWith("'s Party")) {
-            val floor = event.stack.getLore().find { it.startsWith("§7Floor: ") } ?: return
-            val dungeon = event.stack.getLore().find { it.startsWith("§7Dungeon: ") } ?: return
-            val floorNum = floor.split(' ').last().romanToDecimalIfNecessary().toString()
-            val isMasterMode = dungeon.contains("Master Mode")
-
-            event.stackTip = if (floor.contains("Entrance")) {
-                "E"
-            } else if (isMasterMode) {
-                "M${floorNum}"
-            } else {
-                "F${floorNum}"
+                if(config.floorAsStackSize) {
+                    for ((slot, stack) in event.inventoryItems) {
+                        val name = stack.displayName.removeColor()
+                        if (floorTypePattern.matches(name)) {
+                            val floorNum = name.split(" ").last().romanToDecimalIfNecessary().toString()
```

Comment:
since you change everything in here to a pattern, also change the split(" ").last into a pattern

Reply:
how should I go about doing this?

Reply:
".*[IV\\d]+"e
I do not know if roman or decimal that's why both [IV] and \\d